wsclient/src/main/java/org/hpccsystems/ws/client/BaseHPCCWsClient.java

```diff
@@ -59,11 +62,141 @@ public abstract class BaseHPCCWsClient extends DataSingleton
     protected String              initErrMessage         = "";
     protected Version             targetHPCCBuildVersion = null;
     protected Double              targetESPInterfaceVer  = null;
+    protected Boolean             targetsContainerizedHPCC = null;
+
+    public boolean isTargetHPCCContainerized() throws Exception
+    {
+        if (targetsContainerizedHPCC == null)
+        {
+            if (wsconn == null)
+                throw new Exception("BaseHPCCWsClient: Cannot get target HPCC containerized mode, client connection has not been initialized.");
+
+            targetsContainerizedHPCC = getTargetHPCCIsContainerized(wsconn);
+        }
+        return targetsContainerizedHPCC;
+    }
+
+    private boolean getTargetHPCCIsContainerized(Connection conn) throws Exception
+    {
+        if (wsconn == null)
+            throw new Exception("Cannot get target HPCC containerized mode, client connection has not been initialized.");
+
+        String response = wsconn.sendGetRequest("wssmc/getbuildinfo");//throws
+
+        if (response == null || response.isEmpty())
+            throw new Exception("Cannot get target HPCC containerized mode, received empty " + wsconn.getBaseUrl() + " wssmc/getbuildinfo response");
+
+        setUpContainerizedParser();
+
+        Document document = m_XMLParser.parse(new ByteArrayInputStream(response.getBytes(StandardCharsets.UTF_8)));
+
+        NodeList namedValuesList = (NodeList) m_containerizedXpathExpression.evaluate(document,XPathConstants.NODESET);
+        for(int i = 0; i < namedValuesList.getLength(); i++)
+        {
+            Node ithNamedValuePair = namedValuesList.item(i);
+            NodeList nameAndValue = ithNamedValuePair.getChildNodes();
+            if (nameAndValue.getLength() == 2)
+            {
+                String name = null;
+                String value = null;
+                if (nameAndValue.item(0).getNodeName().equalsIgnoreCase("Name"))
+                {
+                     name = nameAndValue.item(0).getFirstChild().getNodeValue();
+                     value = nameAndValue.item(1).getFirstChild().getNodeValue();
+                }
+                else
+                {
+                    name = nameAndValue.item(1).getFirstChild().getNodeValue();
+                    value = nameAndValue.item(0).getFirstChild().getNodeValue();
+                }
+
+                if (name.equals("CONTAINERIZED"))
```

Comment:
Is this guaranteed to be upper case?

Reply:
There's a single line in the esp producing this key: https://github.com/hpcc-systems/HPCC-Platform/blob/89b9799736ef5b9cc469f6ca6a7ad536941895ce/esp/services/ws_smc/ws_smcService.cpp#L2733

I considered equalsignorecase, but thought the extra processing expense wasn't worth it...
but will make the change in case it ever changes in the future as unlikely as it might be